IDEasy not installing latest version for version pattern

From `settings/devon.properties` I have configured the latest Java 17
```
JAVA_VERSION=17*
# ...
```

Now, I want to update the latest version of Java 17:

```
$ ./ideasy.exe install java
Version 17.0.6_10 of tool java is already installed
```

This is incorrect and as you can see the completion is able to suggest the latest stable 17.* version:
![image](https://github.com/devonfw/IDEasy/assets/1923119/597170b5-f483-4ee6-8de8-128d364f8bd4)

Besides our `VersionIdentifier` comparison logic is really incorrect for Java.
I verified this by adding this method to `VersionIdentifierTest` and it is 100% reproducible already at this low level:
```
  @Test
  public void testCompareJavaVersions() {

    VersionIdentifier v21_35 = VersionIdentifier.of("21_35");
    VersionIdentifier v21_0_2_13 = VersionIdentifier.of("21.0.2_13");
    VersionIdentifier v21_0_3_9 = VersionIdentifier.of("21.0.3_9");
    assertThat(v21_35).isLessThan(v21_0_2_13);
    assertThat(v21_0_2_13).isLessThan(v21_0_3_9);
    assertThat(v21_0_3_9).isGreaterThan(v21_35);
  }
```

The test fails with this error:
```
java.lang.AssertionError: 
Expecting actual:
  21_35
to be less than:
  21.0.2_13 
```

Rationale:
https://adoptium.net/download/
The latest release is currently `21.0.3_9` and `21_35` is an old release.
So is bug #1367 not really fixed?

--- a/cli/src/main/java/com/devonfw/tools/ide/version/VersionSegment.java
+++ b/cli/src/main/java/com/devonfw/tools/ide/version/VersionSegment.java
@@ -222,6 +222,12 @@ public class VersionSegment implements VersionObject<VersionSegment> {
     if (!lettersResult.isEqual()) {
       return lettersResult;
     }
+    if (!"_".equals(this.separator) && "_".equals(other.separator)) {
+      return VersionComparisonResult.GREATER;
+    } else if ("_".equals(this.separator) && !"_".equals(other.separator)) {
+      return VersionComparisonResult.LESS;
+    }
+
     if (this.number < other.number) {
       return VersionComparisonResult.LESS;
     } else if (this.number > other.number) {

--- a/cli/src/test/java/com/devonfw/tools/ide/version/VersionIdentifierTest.java
+++ b/cli/src/test/java/com/devonfw/tools/ide/version/VersionIdentifierTest.java
@@ -249,4 +249,15 @@ public class VersionIdentifierTest extends Assertions {
     assertThat(pattern.matches(VersionIdentifier.of("17.0-SNAPSHOT"))).isTrue();
   }
 
+  @Test
+  public void testCompareJavaVersions() {
+
+    VersionIdentifier v21_35 = VersionIdentifier.of("21_35");
+    VersionIdentifier v21_0_2_13 = VersionIdentifier.of("21.0.2_13");
+    VersionIdentifier v21_0_3_9 = VersionIdentifier.of("21.0.3_9");
+    assertThat(v21_35).isLessThan(v21_0_2_13);
+    assertThat(v21_0_2_13).isLessThan(v21_0_3_9);
+    assertThat(v21_0_3_9).isGreaterThan(v21_35);
+  }
+
 }

